components/ILIAS/OpenIdConnect/classes/class.ilOpenIdConnectSettingsGUI.php

```diff
@@ -498,68 +440,269 @@ protected function prepareRoleSelection(bool $a_with_select_option = true): arra
     protected function profile(ilPropertyFormGUI $form = null): void
     {
         $this->checkAccess('read');
-        $this->setSubTabs(self::STAB_PROFILE);
+        $this->chooseMapping();
+        $this->userMapping();
+    }
 
-        if (!$form instanceof ilPropertyFormGUI) {
-            $form = $this->initProfileForm();
+    private function scopes(ilPropertyFormGUI $form = null): void
+    {
+        $this->checkAccess('read');
+        $this->setSubTabs(self::STAB_SCOPES);
+        $url = $this->settings->getProvider();
+        if ($url !== '') {
```

Comment:
IMO we'll have to also check a potentially existing setting regarding the scope validation here, like in `discoverScopesFromServer`. Or am I wrong?

Reply:
I don't think so, this only checks if the config form was filed out, so the quick check if the required field "url" is set, should be enough here.